backend/pom.xml

```diff
@@ -142,6 +142,7 @@
     <dependency>
       <groupId>org.testcontainers</groupId>
       <artifactId>testcontainers</artifactId>
+      <version>${testcontainers.version}</version>
```

Comment:
This is not required. Testcontainers dependency is linked from the parent pom's `<dependencyManagement>` section using the testcontainers BOM.

Reply:
So the culprit here is the inclusion of the `quarkus-bom`, which also imports the `testcontainers-bom`.  The version of Quarkus being used in the backend imports 1.15.2, which is why that version is used there.  

I tried testing exclusion of the `testcontainers-bom` when we import the `quarkus-bom`, but this does not seem to work and we still get the 1.15.2 dependencies.  What does work however is if we import the `testcontainers-bom` inside the backend pom.xml but prior to the `quarkus-bom` and all testcontainer dependencies then follow our specified testcontainers.version.  

Steps:
1. Remove `testcontainers-bom` from the room pom.xml
2. Adjust the backend pom.xml:
```
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.testcontainers</groupId>
        <artifactId>testcontainers-bom</artifactId>
        <version>${testcontainers.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
      <dependency>
        <groupId>${quarkus.platform.group-id}</groupId>
        <artifactId>${quarkus.platform.artifact-id}</artifactId>
        <version>${quarkus.platform.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
  ```
  3. The dependency tree is now:
  ```
  └─$ ./mvnw dependency:tree | grep testcontainers
[INFO] +- org.testcontainers:testcontainers:jar:1.17.6:test
[INFO] +- org.testcontainers:junit-jupiter:jar:1.17.6:test
[INFO] +- io.debezium:debezium-testing-testcontainers:jar:2.2.0-SNAPSHOT:compile
[INFO] +- org.testcontainers:kafka:jar:1.17.6:compile
[INFO] +- org.testcontainers:jdbc:jar:1.17.6:compile
[INFO] |  \- org.testcontainers:database-commons:jar:1.17.6:compile
[INFO] +- org.testcontainers:postgresql:jar:1.17.6:test
[INFO] +- org.testcontainers:mysql:jar:1.17.6:test
[INFO] +- org.testcontainers:mongodb:jar:1.17.6:test
[INFO] \- org.testcontainers:mssqlserver:jar:1.17.6:test
```

Reply:
Can we all agree on this, because without using 1.17.6, we have container issues and so we must explicitly upgrade to this version somehow, either by explicitly overriding the version in all the artifacts listed in the backend pom or by importing the `testcontainers-bom` prior to the `quarkus-bom` within the same maven pom as shown above.

Reply:
Yes, that makes sense. Maybe we should put a detailed comment in the backend pom there to explain why it must be there, if order is important, etc.

Reply:
@Naros @rk3rn3r Added a brief comment and updated the pom files accordingly.